Update asset reference to be more descriptive on behavior and requirements (#1636)
sensu/sensu-docs: content/sensu-go/5.11/reference/assets.md

You can use assets to provide the plugins, libraries, and runtimes you need to automate your monitoring workflows.
Sensu supports runtime assets for [checks][6], [filters][7], [mutators][8], and [handlers][9].

_NOTE: Assets are not required to use Sensu Go in production. Sensu plugins can still be installed using the [sensu-install][32] tool or a [configuration management][33] solution._

## How do assets work?
Assets can be executed by the backend (for handler, filter, and mutator assets), or
by the agent (for check assets). At runtime, the backend or agent sequentially fetches

> assets that appear in the `runtime_assets` attribute of the handler, filter, mutator or check

Reviewer: ```suggestion
assets that appear in the `runtime_assets` attribute of the handler, filter, mutator or check being executed,
```

> and stores them in the backend or agent's local cache. The location of each asset defined in `runtime_assets`

Reviewer: ```suggestion
and stores them in the backend or agent's local cache. The directory path of each asset defined in `runtime_assets`
```

> and stores them in the backend or agent's local cache. The location of each asset defined in `runtime_assets`

Reviewer: ```suggestion
verifies the sha512 checksum, and unpacks them into the backend or agent's local cache directory. The location of each asset defined in `runtime_assets`
```

> is then injected into the `PATH` before handler, filter, mutator or check is executed.

Reviewer: ```suggestion
is then injected into the `PATH` before the handler, filter, mutator or check `command` is executed.
```

Subsequent handler, filter, mutator or check executions look for the asset in the local

> cache and ensure the contents match the configured checksum. The backend or agent's local cache can

Reviewer: ```suggestion
cache and ensure the contents match the configured checksum. The backend or agent's local cache path can
```

be set using the `--cache-dir` flag.

> You can disable assets for an agent using the agent `disable-assets` [configuration flag][30].

Reviewer: ```suggestion
You can disable assets for an agent using the agent `--disable-assets` [configuration flag][30].
```

> You can find use case of a Sensu resource (a check) and an asset in this [example asset with a check][31].

Reviewer 2: Looks like there's a word missing here between find and use.

{{< /language-toggle >}}

> ### Example Asset with a Check

Reviewer 2: ```suggestion
### Example asset with a check
```

Reviewer 2: Sentence case all the things!

{{< language-toggle >}}

{{< highlight yml >}}
---
type: Asset
api_version: core/v2
metadata:
  name: sensu-prometheus-collector_linux_amd64
spec:
  url: https://assets.bonsai.sensu.io/ef812286f59de36a40e51178024b81c69666e1b7/sensu-prometheus-collector_1.1.6_linux_amd64.tar.gz
  sha512: a70056ca02662fbf2999460f6be93f174c7e09c5a8b12efc7cc42ce1ccb5570ee0f328a2dd8223f506df3b5972f7f521728f7bdd6abf9f6ca2234d690aeb3808
  filters:
  - entity.system.os == 'linux'
  - entity.system.arch == 'amd64'
---
type: CheckConfig
api_version: core/v2
metadata:
  name: prometheus_collector
  namespace: default
spec:
  command: "sensu-prometheus-collector -prom-url http://localhost:9090 -prom-query up"
  handlers:

>   - influxdb

Reviewer 2: If I recall correctly, the best practice is to put this under `output_metric_handlers` instead of `handlers`.

Author: Yup! Nice catch, that is true. I'll get that fixed.
